### ISSUE TITLE
Errors

### ISSUE THREAD

**Issue by Lu5ck** [mgm3746/garbagecat#208]
`java -jar garbagecat-4.0.2-SNAPSHOT.jar -p nzgc.log`
[nzgc.zip](https://github.com/user-attachments/files/17818582/nzgc.zip)
[report.txt](https://github.com/user-attachments/files/17818583/report.txt)

**Comment by mgm3746** [comment]
Hi, Thank you for reporting this. I have added additional preprocessing to account for these new patterns. Please test and confirm the fix.

**Comment by mgm3746** [comment]
Hi,

Currently, the ZGC events only track time (implement BlockingEvent).

For example:
https://github.com/mgm3746/garbagecat/blob/main/src/main/java/org/eclipselabs/garbagecat/domain/jdk/unified/ZRelocateStartYoungEvent.java

The ZGC logging is very different than the other collectors. I don't think it's possible to get the metaspace/heap information from typical production logging settings, which is the goal/target for garbagecat.

It could be computed from the Heap before/after information, but I don't consider that something that would be turned on in production because it is too verbose, too much overhead.

IMO, these are not realistic garbage collection settings for production:
gc*=info,heap*=debug,safepoint=info,phases*=debug,gc+stats=debug,stringdedup*=debug,age*=debug,alloc*=debug,metaspace*=info

These settings will fill up storage with massive amounts of useless data. I say "useless" because if there were a production issue, the 1st thing I would say is to remove them all and implement standard logging (to rule out excessive logging being the issue).

Unless/until someone can show otherwise, this is all you need for production:
:gc*,safepoint=info

It's possible some of the debug options could be turned on temporarily for troubleshooting, but in hundreds/thousands of performance issues, I don't recall using any of them. 

Basically, what I have done for this issue is throw all the logging away.

I'm open to adding heap/metaspace information, if someone can show how that is available w/ standard gc logging. Or, explain why the standard logging should be changed to consider other patterns.

**Comment by mgm3746** [comment]
garbagecat will show that information if it's available and being parsed out.

The challenge is that with ZGC, there is no information about the heap and metaspace in the safepoint events. The safepoints are for synchronization only, and collections happen separately, concurrently.

I will look into parsing this out.

**Comment by mgm3746** [comment]
Hi,

I updated to account for the non-safepoint collections with the same handling as I did for Shenandoah. Please test and confirm it is working now.

Thank you for reporting this.

**Comment by mgm3746** [comment]
I believe this is resolved now, but please re-open if that is not correct.
